lib/cli/src/commands/run.rs

```diff
@@ -376,48 +372,46 @@ impl RunWithPathBuf {
     fn run_container(
         &self,
         container: WapmContainer,
-        id: &str,
+        id: Option<&str>,
         args: &[String],
     ) -> Result<(), anyhow::Error> {
-        let mut result = None;
-
-        #[cfg(feature = "wasi")]
-        {
-            if let Some(r) = result {
-                return r;
-            }
-
-            let (store, _compiler_type) = self.store.get_store()?;
-            let mut runner = wasmer_wasi::runners::wasi::WasiRunner::new(store);
-            runner.set_args(args.to_vec());
-            result = Some(if id.is_empty() {
-                runner.run(&container).map_err(|e| anyhow::anyhow!("{e}"))
-            } else {
-                runner
-                    .run_cmd(&container, id)
-                    .map_err(|e| anyhow::anyhow!("{e}"))
-            });
+        let id = id
+            .or_else(|| container.manifest().entrypoint.as_deref())
+            .context("No command specified")?;
+
+        let command = container
+            .manifest()
+            .commands
+            .get(id)
+            .with_context(|| format!("No metadata found for the command, \"{id}\""))?;
+
+        let (store, _compiler_type) = self.store.get_store()?;
+        let mut runner = wasmer_wasi::runners::wasi::WasiRunner::new(store);
+        runner.set_args(args.to_vec());
+        if runner.can_run_command(id, command).unwrap_or(false) {
+            runner
+                .run_cmd(&container, id)
+                .context("WASI runner failed")?;
         }
 
-        #[cfg(feature = "emscripten")]
-        {
-            if let Some(r) = result {
-                return r;
-            }
+        let (store, _compiler_type) = self.store.get_store()?;
+        let mut runner = wasmer_wasi::runners::emscripten::EmscriptenRunner::new(store);
+        runner.set_args(args.to_vec());
+        if runner.can_run_command(id, command).unwrap_or(false) {
+            runner
+                .run_cmd(&container, id)
+                .context("Emscripten runner failed")?;
+        }
 
-            let (store, _compiler_type) = self.store.get_store()?;
-            let mut runner = wasmer_wasi::runners::emscripten::EmscriptenRunner::new(store);
-            runner.set_args(args.to_vec());
-            result = Some(if id.is_empty() {
-                runner.run(&container).map_err(|e| anyhow::anyhow!("{e}"))
-            } else {
-                runner
-                    .run_cmd(&container, id)
-                    .map_err(|e| anyhow::anyhow!("{e}"))
-            });
+        let mut runner = wasmer_wasi::runners::wcgi::WcgiRunner::new(id);
+        runner.config().args(args);
+        if runner.can_run_command(id, command).unwrap_or(false) {
+            runner
+                .run_cmd(&container, id)
+                .context("WCGI runner failed")?;
         }
 
-        result.unwrap_or_else(|| Err(anyhow::anyhow!("neither emscripten or wasi file")))
+        anyhow::bail!("No runner");
```

Comment:
This error is not descriptive. Something like "No WCGI runner was found in the package with the name {X}" is way more helpful. Note: we may also want to specify when a runner is found, but is not wcgi